From 3b4b48b913aa2d3db060ddbec98654a81b910516 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Fri, 28 Nov 2025 10:47:08 +0100
Subject: [PATCH] drm/panel: simple: rework runtime power management

Set late the suspend to prevent a conflict between calling the
"panel unprepare" function and the "panel suspend" function when
executing the "systemctl suspend" command.

Change-Id: I81c89a9aa1b8b58e24f3ff1f4bf9561d00408a87
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/513471
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/panel/panel-simple.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index a708bd1566a3..bc20c684ef28 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -4616,8 +4616,8 @@ static void panel_simple_platform_shutdown(struct platform_device *pdev)
 
 static const struct dev_pm_ops panel_simple_pm_ops = {
 	SET_RUNTIME_PM_OPS(panel_simple_suspend, panel_simple_resume, NULL)
-	SET_SYSTEM_SLEEP_PM_OPS(pm_runtime_force_suspend,
-				pm_runtime_force_resume)
+	SET_LATE_SYSTEM_SLEEP_PM_OPS(pm_runtime_force_suspend,
+				     pm_runtime_force_resume)
 };
 
 static struct platform_driver panel_simple_platform_driver = {
-- 
2.53.0

